respy/simulate.py

```diff
@@ -205,33 +207,83 @@ def simulate(
 
         wages = state_space.get_attribute_from_period("wages", period)
         nonpecs = state_space.get_attribute_from_period("nonpecs", period)
-        index_to_choice_set = state_space.get_attribute_from_period(
-            "dense_key_to_choice_set", period
+        index_to_complex = state_space.get_attribute_from_period(
+            "dense_key_to_complex", period
         )
         continuation_values = state_space.get_continuation_values(period=period)
 
         current_df_extended = _simulate_single_period(
             current_df,
-            index_to_choice_set,
+            index_to_complex,
             wages,
             nonpecs,
             continuation_values,
             optim_paras=optim_paras,
+            options=options,
         )
 
-        data.append(current_df_extended.copy(deep=True))
+        if optim_paras["exogenous_processes"]:
+            data.append(
+                current_df_extended.copy(deep=True).drop(
+                    "dense_key_next_period", axis=1
+                )
+            )
+        else:
+            data.append(current_df_extended.copy(deep=True))
 
         if is_n_step_ahead and period != n_simulation_periods - 1:
             current_df_extended = current_df_extended.reset_index()
             df = apply_law_of_motion_for_core(current_df_extended, optim_paras)
             state_space_columns = create_state_space_columns(optim_paras)
+
+            if optim_paras["exogenous_processes"]:
```

Comment:
Very minor: Could we maybe consider a solution without the switch "if" statement? Maybe just writing a function called apply_law_of_motion_for_dense where the operation is done when df contains the column dense key next period. I think that is better practice.

Reply:
Done!